{"bug_class": "PROVER", "kind": "ERROR", "bug_type": "RESOURCE_LEAK", "qualifier": "resource of type `redis.clients.jedis.JedisCluster` acquired to `jedisCluster` by call to `new()` at line 49 is not released after line 49.", "severity": "HIGH", "visibility": "user", "line": 49, "column": -1, "procedure": "void RedisLimitTest.init()", "procedure_id": "com.crossoverjie.distributed.limit.RedisLimitTest.init():void.ec403d3d784ce6d791f2238f1fa93256", "procedure_start_line": 47, "file": "src/test/java/com/crossoverjie/distributed/limit/RedisLimitTest.java", "bug_trace": [{"level": 0, "filename": "src/test/java/com/crossoverjie/distributed/limit/RedisLimitTest.java", "line_number": 47, "column_number": -1, "description": "start of procedure init()"}, {"level": 0, "filename": "src/test/java/com/crossoverjie/distributed/limit/RedisLimitTest.java", "line_number": 48, "column_number": -1, "description": "Skipping HostAndPort(...): unknown method"}, {"level": 0, "filename": "src/test/java/com/crossoverjie/distributed/limit/RedisLimitTest.java", "line_number": 49, "column_number": -1, "description": "Skipping JedisCluster(...): unknown method"}], "key": "RedisLimitTest.java|init|RESOURCE_LEAK", "node_key": "8917bc3fd71ad534fa2589715e55c5cb", "hash": "79be504c11597d8288be16c5c99dd728", "bug_type_hum": "Resource Leak", "censored_reason": ""}